scripts/build-tools.js

```diff
@@ -1,24 +1,33 @@
 const { getData } = require('./tools/extract-tools-github');
 const { convertTools } = require('./tools/tools-object');
 const { combineTools } = require('./tools/combine-tools');
-const manualTools = require('../config/tools-manual.json')
-
 const fs = require('fs');
 const { resolve } = require('path');
 
-const buildTools = async () => {
+const buildTools = async (automatedToolsPath, manualToolsPath, toolsPath, tagsPath) => {
   try {
     let githubExtractData = await getData();
     let automatedTools = await convertTools(githubExtractData);
+    
     fs.writeFileSync(
-      resolve(__dirname, '../config', 'tools-automated.json'),
+      automatedToolsPath,
       JSON.stringify(automatedTools, null, '  ')
     );
-    await combineTools(automatedTools, manualTools);
+    
+    await combineTools(automatedTools, require(manualToolsPath), toolsPath, tagsPath);
   } catch (err) {
-    console.log(err);
-    throw err
+    throw new Error(`An error occurred while building tools: ${err.message}`);
   }
 };
 
-buildTools();
\ No newline at end of file
+/* istanbul ignore next */
+if (require.main === module) {
```

Comment:
Why we have added this check here ?

Reply:
It is to check whether the script is called directly or imported into another module. Without that check, if we run the tests, this function call inside the script will also be executed, which we don't want.